src/controllers/session.js

```diff
@@ -0,0 +1,67 @@
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+
+const Users = require('../models/User.js')
+const roles = require('../constants/roles.js')
+
+const sessionController = module.exports = {}
+
+sessionController.register = (req, res, next) => {
+  Users.find({ username: req.body.username }).then(user => {
+    if (user.length === 0) {
+      return bcrypt.hash(req.body.password, 12)
+    }
+
+    res.boom.badRequest('username already exists')
+  }).then(passwordHash => {
+    const { englishName, arabicName, email, username, imageUrl } = req.body
+    const en = englishName && { name: englishName }
+    const ar = arabicName && { name: arabicName }
+    return Users.create(Object.assign(
+      {
+        username: username,
+        password: passwordHash,
+        role: roles.BASIC,
+        email: email,
+        imageUrl: imageUrl
+      },
+      { ar },
+      { en }
+    ))
+  }).then(user => {
+    res.send('registered!')
```

Comment:
Yeah I mean this was just a place holder for now, I guess yeah a redirect makes sense, not sure what to as atm we don't have any views to redirect to. Maybe `GET /apps` when it exists? But leave it like this for this pull? I can add a comment